src/main/java/org/apache/sysds/runtime/instructions/fed/CtableFEDInstruction.java

```diff
@@ -0,0 +1,331 @@
+				prevExtend.copy(0, prev.getNumRows()-1, 0, prev.getNumColumns()-1, prev, true);
+
+				MatrixBlock  intersect = curr.binaryOperationsInPlace(new BinaryOperator(And.getAndFnObject()), prevExtend);
+				if(intersect.getNonZeros() != 0)
+					fedOutput = false;
+				prev = sliced;
+			}
+		}
+		catch(Exception e) {
+			e.printStackTrace();
+		}
+		return fedOutput;
+	}
+
+
+	private void setFedOutput(FederatedRequest fr, MatrixObject mo1, MatrixObject out, Long[] dims1, Long[] dims2, long fedSize) {
+		long d1 = Collections.max(Arrays.asList(dims1), Long::compare);
+		long d2 = Collections.max(Arrays.asList(dims1), Long::compare);
+
+		FederationMap fedMap = mo1.getFedMapping();
+		// modify fed ranges
+		for(int i = 0; i < fedMap.getFederatedRanges().length; i++) {
+			fedMap.getFederatedRanges()[i].setBeginDim(0, i == 0 ? 0 : fedMap.getFederatedRanges()[i-1].getEndDims()[0]);
+			fedMap.getFederatedRanges()[i].setEndDim(0, dims1[i]);
+
+			fedMap.getFederatedRanges()[i].setBeginDim(1, i == 0 ? 0 : fedMap.getFederatedRanges()[i-1].getBeginDims()[1]);
+			fedMap.getFederatedRanges()[i].setEndDim(1, dims2[i]);
+		}
+
+		// set output
+		out.getDataCharacteristics().set(d1, d2, (int) mo1.getBlocksize(), mo1.getNnz());
+		out.setFedMapping(fedMap.copyWithNewID(fr.getID()));
+
+		long varID = FederationUtils.getNextFedDataID();
+		out.getFedMapping().mapParallel(varID, (range, data) -> {
+			try {
+				FederatedResponse response = data.executeFederatedOperation(new FederatedRequest(
+					FederatedRequest.RequestType.EXEC_UDF, -1,
+					new SliceOutput(data.getVarID(), fedSize))).get();
+				if(!response.isSuccessful())
+					response.throwExceptionFromResponse();
+			}
+			catch(Exception e) {
+				throw new DMLRuntimeException(e);
+			}
+			return null;
+		});
+	}
+
+	private MatrixBlock aggResult(Future<FederatedResponse>[] ffr) {
+		MatrixBlock resultBlock = new MatrixBlock(1, 1, 0);
+		int dim1 = 0, dim2 = 0;
+		for(int i = 0; i < ffr.length; i++) {
+			try {
+				MatrixBlock mb = ((MatrixBlock) ffr[i].get().getData()[0]);
+				dim1 = mb.getNumRows()  > dim1 ? mb.getNumRows() : dim1;
+				dim2 = mb.getNumColumns()  > dim2 ? mb.getNumColumns() : dim2;
+
+				// set next and prev to same output dimensions
+				MatrixBlock prev = new MatrixBlock(dim1, dim2, 0.0);
+				prev.copy(0, resultBlock.getNumRows()-1, 0, resultBlock.getNumColumns()-1, resultBlock, true);
+
+				MatrixBlock next = new MatrixBlock(dim1, dim2, 0.0);
+				next.copy(0, mb.getNumRows()-1, 0, mb.getNumColumns()-1, mb, true);
+
+				// add worker results
+				BinaryOperator plus = InstructionUtils.parseBinaryOperator("+");
+				resultBlock = prev.binaryOperationsInPlace(plus, next);
+			}
+			catch(Exception e) {
+				e.printStackTrace();
+			}
+		}
+		return resultBlock;
+	}
+
+	private Long[] getOutputDimension(ExecutionContext ec, CPOperand inOp, CPOperand outOp, FederatedRange[] federatedRanges) {
+		MatrixObject in = ec.getMatrixObject(inOp);
```

Comment:
You already loaded this input matrix in the processInstruction method. Why don't you just pass that object to this method instead of loading it again?